tests/integration/cmor/_fixes/icon/test_icon.py

```diff
@@ -0,0 +1,410 @@
+    assert cube.var_name == 'tas'
+    assert cube.standard_name == 'air_temperature'
+    assert cube.long_name == 'Near-Surface Air Temperature'
+    assert cube.units == 'K'
+
+    check_time(cube)
+    check_lat_lon(cube)
+    assert cube.coords('height')
+    height = cube.coord('height')
+    assert height.var_name == 'height'
+    assert height.standard_name == 'height'
+    assert height.long_name == 'height'
+    assert height.units == 'm'
+    assert height.attributes == {'positive': 'up'}
+    np.testing.assert_allclose(height.points, [2.0])
+    assert height.bounds is None
+
+
+# Test uas (for height10m coordinate)
+
+
+def test_get_uas_fix():
+    """Test getting of fix."""
+    fix = Fix.get_fixes('ICON', 'ICON', 'Amon', 'uas')
+    assert fix == [AllVars(None)]
+
+
+def test_uas_fix(cubes_2d):
+    """Test fix."""
+    fix = get_allvars_fix('Amon', 'uas')
+    fixed_cubes = fix.fix_metadata(cubes_2d)
+
+    assert len(fixed_cubes) == 1
+    cube = fixed_cubes[0]
+    assert cube.var_name == 'uas'
+    assert cube.standard_name == 'eastward_wind'
+    assert cube.long_name == 'Eastward Near-Surface Wind'
+    assert cube.units == 'm s-1'
+
+    check_time(cube)
+    check_lat_lon(cube)
+    assert cube.coords('height')
+    height = cube.coord('height')
+    assert height.var_name == 'height'
+    assert height.standard_name == 'height'
+    assert height.long_name == 'height'
+    assert height.units == 'm'
+    assert height.attributes == {'positive': 'up'}
+    np.testing.assert_allclose(height.points, [10.0])
+    assert height.bounds is None
+
+
+# Test fix with regular grid and 2D latitudes and longitude
+
+
+def test_regular_grid_fix(cubes_regular_grid):
+    """Test fix."""
+    fix = get_allvars_fix('Amon', 'tas')
+    fixed_cubes = fix.fix_metadata(cubes_regular_grid)
+
+    assert len(fixed_cubes) == 1
+    cube = fixed_cubes[0]
+    assert cube.var_name == 'tas'
+    assert cube.standard_name == 'air_temperature'
+    assert cube.long_name == 'Near-Surface Air Temperature'
+    assert cube.units == 'K'
+
+    assert cube.coords('latitude', dim_coords=True, dimensions=0)
+    assert cube.coords('longitude', dim_coords=True, dimensions=1)
+    assert cube.coords('height', dim_coords=False, dimensions=())
+
+
+def test_2d_lat_lon_grid_fix(cubes_2d_lat_lon_grid):
+    """Test fix."""
+    fix = get_allvars_fix('Amon', 'tas')
+    fixed_cubes = fix.fix_metadata(cubes_2d_lat_lon_grid)
+
+    assert len(fixed_cubes) == 1
+    cube = fixed_cubes[0]
+    assert cube.var_name == 'tas'
+    assert cube.standard_name == 'air_temperature'
+    assert cube.long_name == 'Near-Surface Air Temperature'
+    assert cube.units == 'K'
+
+    assert cube.coords('latitude', dim_coords=False, dimensions=(0, 1))
+    assert cube.coords('longitude', dim_coords=False, dimensions=(0, 1))
+    assert cube.coords('height', dim_coords=False, dimensions=())
+
+
+# Test fix with empty standard_name
+
+
+def test_empty_standard_name_fix(cubes_2d):
```

Comment:
I don't understand the point of this test, less since obviously there is a correct standard name for tas. Could you explain?

Reply:
To test these lines:

https://github.com/ESMValGroup/ESMValCore/blob/df4458f4079c6b031254d8d30c589bd009a820b2/esmvalcore/cmor/_fixes/icon/icon.py#L58-L61

I know that there is a standard name for tas, this is mainly done to please codecov so that every line is tested.

Reply:
Is there an actual example of a variable with no standard name? That might be a better test. Otherwise, I suggest adding a comment.

Reply:
I am not aware of an actual use case, but in the checks we treat `None` and `''` equally (see e.g.,

https://github.com/ESMValGroup/ESMValCore/blob/d587d0c245ac52cf32326d9ca5a46bc53b982b90/esmvalcore/cmor/check.py#L400-L401

so I figure it's best if we do the same here. `''` is not a valid standard name and `iris` will crash if we try to set that.

Where do you want me to put the comment? In the actual code or the test?

Reply:
I was thinking of a comment in the test to the effect of "We know that tas has a standard name, but this being native model output there may be variables with no standard name. The code is designed to handle this gracefully and here we test it with an artificial, but realistic case."